src/main/java/emissary/client/HTTPConnectionFactory.java

```diff
@@ -86,16 +90,17 @@ public class HTTPConnectionFactory {
     private static final String HTTPS = "https";
     private static final String FILE_PRE = "file://";
     private static final Pattern ENV_VARIABLE_PATTERN = Pattern.compile("\\$\\{(\\w+)}");
+    private static final Pattern CERT_PATTERN = Pattern.compile(
```

Comment:
CodeQL flags this Pattern as risky, although this is largely mitigated since we control the input files that this will be matched against.  A bounded Pattern such as this would be cleaner.  I chose the minimum and maximum lengths somewhat arbitrarily, but they're conservatively broad while preventing runaway scans.

```
private static final Pattern CERT_PATTERN = Pattern.compile(
"(?:^-----BEGIN CERTIFICATE-----$)([^-]{64,4096})(?:^-----END CERTIFICATE-----$)", Pattern.MULTILINE);
```

Reply:
Updated again to included a named capture group.

Reply:
Ideally, yes.  My snippet was lazier and deferred validation until trying to construct the actual X509Certficate, but it's better to constrain the characters in the Pattern itself.   I think I was just annoyed that I couldn't successfully constrain the length of individual Base64-encoded lines within a multi-line capture group, so I overcorrected when simplifying.   Your change is better.